### PR TITLE
Force install dependencies in old Debian with expired keys

### PR DESCRIPTION
**Pull request by rdner**
## What does this PR do?

This change is the same as https://github.com/elastic/beats/pull/33915 but for the rest of the Beats.

## Why is it important?

We cannot build snapshots right now:

```
06:29:52 Status: Downloaded newer image for docker.elastic.co/beats-dev/golang-crossbuild:1.18.10-main-debian8
06:29:58 >> Building using: cmd='build/mage-linux-amd64 golangCrossBuild', env=[CC=gcc, CXX=g++, GOARCH=386, GOARM=, GOOS=linux, PLATFORM_ID=linux-386]
06:29:58 >> Building using: cmd='build/mage-linux-amd64 golangCrossBuild', env=[CC=gcc, CXX=g++, GOARCH=amd64, GOARM=, GOOS=linux, PLATFORM_ID=linux-amd64]
06:30:05 W: GPG error: http://archive.debian.org/ jessie Release: The following signatures were invalid: KEYEXPIRED 1587841717
06:30:05 W: Failed to fetch http://security.debian.org/debian-security/dists/jessie/updates/main/binary-amd64/Packages  404  Not Found [IP: 151.101.194.132 80]
06:30:05 
06:30:05 W: Failed to fetch http://security.debian.org/debian-security/dists/jessie/updates/main/binary-i386/Packages  404  Not Found [IP: 151.101.194.132 80]
06:30:05 
06:30:05 E: Some index files failed to download. They have been ignored, or old ones used instead.
06:30:05 Error: running "apt-get update" failed with exit code 100
06:30:05 Error: failed building for linux/386: exit status 100
06:30:05 failed building for linux/386: exit status 100
06:30:05 W: GPG error: http://archive.debian.org/ jessie Release: The following signatures were invalid: KEYEXPIRED 1587841717
06:30:05 W: Failed to fetch http://security.debian.org/debian-security/dists/jessie/updates/main/binary-amd64/Packages  404  Not Found [IP: 151.101.66.132 80]
06:30:05 
06:30:05 W: Failed to fetch http://security.debian.org/debian-security/dists/jessie/updates/main/binary-i386/Packages  404  Not Found [IP: 151.101.66.132 80]
06:30:05 
06:30:05 E: Some index files failed to download. They have been ignored, or old ones used instead.
06:30:05 Error: running "apt-get update" failed with exit code 100
06:30:05 Error: failed building for linux/amd64: exit status 100
06:30:05 failed building for linux/amd64: exit status 100
```

https://internal-ci.elastic.co/job/elastic+unified-release+master+snapshot-multijob-7.17/361/display/redirect

## How to test locally

```sh
mage package
```

### CODE REVIEW

**Review by cmacknz — APPROVED** [review]

**Comment by rdner** [comment]
It's broken, I'm trying out more options.

**Comment by rdner** [comment]
I have these errors when I just run `apt update` on raw `Debian Jessie (8)`:

```
root@3b6c3be95753:/etc/apt# apt update
Ign http://security.debian.org jessie/updates InRelease
Ign http://security.debian.org jessie/updates Release.gpg
Ign http://security.debian.org jessie/updates Release 
Ign http://ftp.us.debian.org jessie InRelease
Ign http://ftp.us.debian.org jessie Release.gpg
Ign http://ftp.us.debian.org jessie Release
Err http://security.debian.org jessie/updates/main armhf Packages
  404  Not Found [IP: 151.101.2.132 80]
Err http://security.debian.org jessie/updates/contrib armhf Packages
  404  Not Found [IP: 151.101.2.132 80]
Err http://security.debian.org jessie/updates/non-free armhf Packages
  404  Not Found [IP: 151.101.2.132 80]
Err http://ftp.us.debian.org jessie/main armhf Packages
  404  Not Found [IP: 208.80.154.139 80]
Err http://ftp.us.debian.org jessie/contrib armhf Packages
  404  Not Found [IP: 208.80.154.139 80]
Err http://ftp.us.debian.org jessie/non-free armhf Packages
  404  Not Found [IP: 208.80.154.139 80]
W: Failed to fetch http://security.debian.org/dists/jessie/updates/main/binary-armhf/Packages  404  Not Found [IP: 151.101.2.132 80]

W: Failed to fetch http://security.debian.org/dists/jessie/updates/contrib/binary-armhf/Packages  404  Not Found [IP: 151.101.2.132 80]

W: Failed to fetch http://security.debian.org/dists/jessie/updates/non-free/binary-armhf/Packages  404  Not Found [IP: 151.101.2.132 80]

W: Failed to fetch http://ftp.us.debian.org/debian/dists/jessie/main/binary-armhf/Packages  404  Not Found [IP: 208.80.154.139 80]

W: Failed to fetch http://ftp.us.debian.org/debian/dists/jessie/contrib/binary-armhf/Packages  404  Not Found [IP: 208.80.154.139 80]

W: Failed to fetch http://ftp.us.debian.org/debian/dists/jessie/non-free/binary-armhf/Packages  404  Not Found [IP: 208.80.154.139 80]

E: Some index files failed to download. They have been ignored, or old ones used instead.
```

Looks like the official repositories are not available anymore.

**Comment by rdner** [comment]
I fixed the source.list (kind of) but I still have this problem to solve:

```
The following packages have unmet dependencies:
 libgcrypt20:i386 : Depends: libc6:i386 (>= 2.15) but it is not installable
                    Depends: libgpg-error0:i386 (>= 1.14) but it is not going to be installed
 libsystemd0:i386 : PreDepends: libc6:i386 (>= 2.17) but it is not installable
                    PreDepends: libgcc1:i386 (>= 1:4.1.1) but it is not installable
                    PreDepends: liblzma5:i386 (>= 5.1.1alpha+20120614) but it is not installable
                    PreDepends: libselinux1:i386 (>= 1.32) but it is not installable
E: Unable to correct problems, you have held broken packages.
```

File: dev-tools/mage/pkgdeps.go [hunk 1]
```diff
@@ -117,22 +120,64 @@ func installDependencies(arch string, pkgs ...string) error {
...
+	if err != nil {
+		return fmt.Errorf("error while editing the repositories: %w", err)
 	}
 
-	params := append([]string{"install", "-y", "--force-yes",
-		"--no-install-recommends",
+	// TODO: This is only for debian 7 and should be removed when move to a newer OS. This flag is
+	// going to be used unnecessary when building using non-debian7 images
+	// (like when making the linux/arm binaries) and we should remove it soonish.
+	// See https://github.com/elastic/beats/v7/issues/11750 for more details.
```
Comment (faec) [inline dev-tools/mage/pkgdeps.go:131]: URL is broken (remove `v7/`)

Comment (rdner) [reply dev-tools/mage/pkgdeps.go:131]: Thanks! Will do, it was copy pasted from here https://github.com/elastic/beats/blob/2d32a6e6b541ce8b5a7c377e99bc0243911b40ab/x-pack/auditbeat/magefile.go#L221

**Comment by rdner** [comment]
This should be fixed by https://github.com/elastic/beats/pull/34939

**Pull request closed without merging**